src/main/java/io/apicurio/datamodels/core/validation/rules/required/MissingApiKeySchemeParamLocationRule.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.datamodels.core.validation.rules.required;
+
+import io.apicurio.datamodels.core.Constants;
+import io.apicurio.datamodels.core.models.DocumentType;
+import io.apicurio.datamodels.core.models.common.SecurityScheme;
+import io.apicurio.datamodels.core.validation.ValidationRuleMetaData;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+public class MissingApiKeySchemeParamLocationRule extends RequiredPropertyValidationRule {
+
+    /**
+     * Constructor.
+     * 
+     * @param ruleInfo
+     */
+    public MissingApiKeySchemeParamLocationRule(ValidationRuleMetaData ruleInfo) {
+        super(ruleInfo);
+    }
+
+    /**
+     * @see io.apicurio.datamodels.combined.visitors.CombinedAllNodeVisitor#visitSecurityScheme(io.apicurio.datamodels.core.models.common.SecurityScheme)
+     */
+    @Override
+    public void visitSecurityScheme(SecurityScheme node) {
+        if (node.ownerDocument().getDocumentType() == DocumentType.asyncapi2) {
+            if (equals(node.type, "apiKey")) {
+                this.requirePropertyWhen(node, Constants.PROP_IN, Constants.PROP_TYPE, "apiKey", map());
+            } else if (equals(node.type, "httpApiKey")) {
+                this.requirePropertyWhen(node, Constants.PROP_IN, Constants.PROP_TYPE, "httpApiKey", map());
+            }
```

Comment:
I think this should instead be something like this:

```java
if (equals(node.type, "apiKey") || equals(node.type, "httpApiKey")) {
    this.requireProperty(node, Constants.PROP_IN, map());
}
```

The way you have it now will work, but you're basically doing the "apiKey" or "httpApiKey" comparison twice.  Once on line 44/46 and then again in the `requirePropertyWhen` method.

Reply:
@EricWittmann yep I see that now.. thanks